app/Http/Controllers/WikiController.php

```diff
@@ -27,26 +28,56 @@ class WikiController extends Controller
      *
      * ### Response Format
      *
-     * Returns [WikiPage](#wikipage) if the content is a wiki page; a binary blob, otherwise.
+     * Returns [WikiPage](#wikipage).
      *
      * @urlParam page The path name of the wiki page.
      */
-    public function show($path = null)
+    public function show($locale = null, $path = null)
     {
-        if ($path === null) {
-            return ujs_redirect(wiki_url());
+        // redirect to default page if missing all parameters
+        if ($locale === null) {
+            return ujs_redirect(wiki_url(null, $this->locale()));
         }
 
+        // if images slip through the markdown processing, redirect them to the correct place
         if (OsuWiki::isImage($path)) {
-            return $this->showImage($path);
+            return ujs_redirect(route('wiki.image', concat_path([$locale === 'images' ? null : $locale, $path])));
```

Comment:
Shouldn't this also skip adding the locale to the path for images if it's a valid locale?